### PR TITLE
Update Docs Role Handbook based on 1.27

### PR DESCRIPTION
**Pull request by Rishit-dagli**
#### What type of PR is this:
/kind cleanup
/kind documentation

#### What this PR does / why we need it:
Updates the Docs Role Handbook with up-to-date workflows and fixes some minor typos.

#### Special notes for your reviewer:
Follow-up to https://github.com/kubernetes/sig-release/pull/2302 and https://github.com/kubernetes/sig-release/pull/2252

### CODE REVIEW

**Comment by Rishit-dagli** [comment]
/cc @divya-mohan0209 @sftim 

Added the note about CLA issues we faced in this release, and added insights based on our Slack thread, could you also PTAL at this. Here's the commit introducing the change https://github.com/kubernetes/sig-release/pull/2311/commits/54a3d16a676a825092a61a3525f376e5be68b284 .

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (kcmartin) [inline release-team/role-handbooks/docs/Release-Timeline.md:587]:
```suggestion
You may need to change `hugo.toml`, instead of `config.toml`, starting from the 1.26 release.
```

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (kcmartin) [inline release-team/role-handbooks/docs/Release-Timeline.md:586]:
```suggestion
Update the `config.toml` files for the 4 previous releases. You need to open 4 separate PRs, because each release has its own `release-` branch.
```

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (kcmartin) [inline release-team/role-handbooks/docs/Release-Timeline.md:589]:
```suggestion
for the previous release targeting `main`, and once you create the `release-[previous-release]` branch you should then change the
```

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (kcmartin) [inline release-team/role-handbooks/docs/Release-Timeline.md:588]:
```suggestion
There would not be a `release-[previous-release]` branch, so you should make the pull request updating the site configuration
```

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (kcmartin) [inline release-team/role-handbooks/docs/Release-Timeline.md:747]:
```suggestion
> Note: In 1.28 the Docs Release Team had failing CLA checks while merging the integration branch, in which case they made a decision to not change history at all. Rather, they merged the integration branch even with failing CLA checks since the commit did pass CLA when it was merged to `dev-1.28`, and the docs team did not want to fix this by altering history due to the side effects of doing so.
```

**Review by kcmartin — CHANGES_REQUESTED** [review]

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (Rishit-dagli) [reply release-team/role-handbooks/docs/Release-Timeline.md:589]: Thanks!

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (Rishit-dagli) [reply release-team/role-handbooks/docs/Release-Timeline.md:747]: Thanks!

File: release-team/role-handbooks/docs/Release-Timeline.md
Comment (Rishit-dagli) [reply release-team/role-handbooks/docs/Release-Timeline.md:587]: Sounds better

**Review by kcmartin — APPROVED** [review]

**Comment by Rishit-dagli** [comment]
We shouldn't merge this unless we get a review from everyone at the docs retrospective
/hold

**Comment by Rishit-dagli** [comment]
This was supposed to be on hold until after the SIG Docs retro, and I also added a hold but for some reason, Prow still merged it 😓 .

**Comment by sftim** [comment]
I think Prow's had a busy day&hellip;

**Comment by sftim** [comment]
BTW, to _ensure_ (pretty much) that a PR doesn't merge and that Prow notices, mark the PR as draft using GitHub's API.

**Pull request merged**
